Sources/Datadog/RUM/RUMMonitor/Scopes/RUMUserActionScope.swift

```diff
@@ -0,0 +1,180 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import Foundation
+
+internal class RUMUserActionScope: RUMScope {
+    struct Constants {
+        /// If no activity is observed within this period in a discrete (discontinous) User Action, it is condiered ended.
+        /// The activity may be i.e. due to Resource started loading.
+        static let discreteActionTimeoutDuration: TimeInterval = 0.1 // 100 milliseconds
+        /// Maximum duration of a continuous User Action. If it gets exceeded, a new session is started.
+        static let continuousActionMaxDuration: TimeInterval = 10 // 10 seconds
+    }
+
+    // MARK: - Initialization
+
+    // TODO: RUMM-597: Consider using `parent: RUMContextProvider`
+    private unowned let parent: RUMScope
+    private let dependencies: RUMScopeDependencies
+
+    /// The type of this User Action.
+    internal let actionType: RUMUserActionType
+    /// User Action attributes.
+    private(set) var attributes: [AttributeKey: AttributeValue]
+
+    /// This User Action's UUID.
+    internal let actionUUID: RUMUUID
+    /// The start time of this User Action.
+    private var actionStartTime: Date
+    /// Tells if this action is continuous over time, like "scroll" (or discrete, like "tap").
+    internal let isContinuous: Bool
+    /// Time of the last RUM activity noticed by this User Action (i.e. Resource loading).
+    private var lastActivityTime: Date
+
+    /// Number of Resources started during this User Action's lifespan.
+    private var resourcesCount: UInt = 0
+    /// Number of Errors occured during this User Action's lifespan.
+    private var errorsCount: UInt = 0
+    /// Number of Resources that started but not yet ended during this User Action's lifespan.
+    private var activeResourcesCount: Int = 0
+
+    init(
+        parent: RUMScope,
+        dependencies: RUMScopeDependencies,
+        actionType: RUMUserActionType,
+        attributes: [AttributeKey: AttributeValue],
+        startTime: Date,
+        isContinuous: Bool
+    ) {
+        self.parent = parent
+        self.dependencies = dependencies
+        self.actionType = actionType
+        self.attributes = attributes
+        self.actionUUID = dependencies.rumUUIDGenerator.generateUnique()
+        self.actionStartTime = startTime
+        self.isContinuous = isContinuous
+        self.lastActivityTime = startTime
+    }
+
+    // MARK: - RUMScope
+
+    var context: RUMContext {
+        return parent.context
```

Comment:
The child scope cannot exist without the parent and it should not own the parent - this is why we use `unowned`. If using `weak`, this would mean: the child scope can exist without its parent, which is not allowed in our scopes hierarchy.